datadog/resource_datadog_synthetics_test_.go

```diff
@@ -3783,3 +3791,71 @@ func validateSyntheticsAssertionOperator(val interface{}, key string) (warns []s
 	}
 	return
 }
+
+func getConfigCertificate(rawConfig cty.Value, stepIndex int) (*string, *string) {
+
+	basePath := cty.GetAttrPath("api_step").
+		Index(cty.NumberIntVal(int64(stepIndex))).
+		GetAttr("request_client_certificate").
+		Index(cty.NumberIntVal(0))
+
+	// Construct paths to the cert and key content
+	certPath := basePath.
+		GetAttr("cert").
+		Index(cty.NumberIntVal(0)).
+		GetAttr("content")
+
+	keyPath := basePath.
+		GetAttr("key").
+		Index(cty.NumberIntVal(0)).
+		GetAttr("content")
+
+	// Apply paths to retrieve the cert and key
+	certValue, err := certPath.Apply(rawConfig)
+	if err != nil || !certValue.IsKnown() || certValue.IsNull() {
+		return nil, nil
+	}
+
+	keyValue, err := keyPath.Apply(rawConfig)
+	if err != nil || !keyValue.IsKnown() || keyValue.IsNull() {
+		return nil, nil
+	}
+
+	// Convert certValue and keyValue to strings
+	certString := certValue.AsString()
+	keyString := keyValue.AsString()
+
+	return &certString, &keyString
+}
+
+func overrideStateCertificate(requestClientCertificates []interface{}, configCert, configKey string) error {
+
+	if len(requestClientCertificates) == 0 {
+		return fmt.Errorf("requestClientCertificates is empty")
+	}
+	requestClientCertificate, ok := requestClientCertificates[0].(map[string]interface{})
+	if !ok {
```

Comment:
It's idiomatic in Go to have the assignments in the `if` statement:
```suggestion
	if requestClientCertificate, ok := requestClientCertificates[0].(map[string]interface{}); !ok {
```

Reply:
Doing so means requestClientCertificate is undefined in the rest of the code I think... It's only defined in the if, and I wanted to avoid having nested ifs